src/shared/netif-naming-scheme.h

```diff
@@ -43,6 +43,7 @@ typedef enum NamingSchemeFlags {
         NAMING_DEVICETREE_ALIASES        = 1 << 15, /* Generate names from devicetree aliases */
         NAMING_USB_HOST                  = 1 << 16, /* Generate names for usb host */
         NAMING_SR_IOV_R                  = 1 << 17, /* Use "r" suffix for SR-IOV VF representors */
+        NAMING_DEVICETREE_ETH_PORTS      = 1 << 18, /* Allow generate names from ethernet-ports devicetree aliases */
```

Comment:
not a fan of needless abbreviations. just spell out ETHERNET

Reply:
the idea of those flags is that you can control with them the naming rules, i.e. enable/disable naming rules added in later versions. Hence, a flag that can be set or not set, but has exactly zero effect is useless. i.e. the whole code you are adding should be enclosed in a condition check if the flag is already set. If not, thing should operate like they used to operate.

Reply:
All right got the point, thanks for explaining. 

I added this check

```c
    if (naming_scheme_has(NAMING_DEVICETREE_ETHERNET_PORTS)) {
                /* Check if there are 'ethernet-ports' child nodes of ethernet */
                r = sd_device_new_child(&eth_ports_dev, ofnode_dev, "ethernet-ports");
                if (r < 0)
```

so if this naming is not being used `eth_ports_dev` will be NULL and the if statement below will also fail.